### PR TITLE
Restore tenant configuration for Loki forwarder.

### PR DESCRIPTION
**Pull request by alanconway**
This configuration will not be required for our default Loki instance, but may be required for custom Loki instances.

/cc @periklis
/assign @jcantrill

### CODE REVIEW

File: enhancements/cluster-logging/forward-to-loki.md [hunk 1]
```diff
@@ -148,6 +148,12 @@ The following optional output fields are Loki-specific:
   Example: `kubernetes.labels.foo` => `kubernetes_labels_foo`.\
   **Note**: `kubernetes.host` is *always* be included, even if not requested.
   It is required to ensure ordered label streams.
+- `tenantKey`: (string, default=`"kubernetes.namespaceName"`) \
```
Comment (periklis) [inline enhancements/cluster-logging/forward-to-loki.md:151]: We don't provide a default for in-cluster.

Comment (periklis) [reply enhancements/cluster-logging/forward-to-loki.md:151]: In addition do we really want to define as tenant key a label key or simply leave it as arbitrary data?

Comment (alanconway) [reply enhancements/cluster-logging/forward-to-loki.md:156]: A fixed tenant ID limits us to static tenants pre-defined by the log admin based on input filters. This has turned out to be inadequate for other cases (JSON, cloudwatch groups etc.), where the logging system needs to respond to new label values or new namespaces appearing in the cluster without a manual re-config each time. I think the same applies to tenants - you can't implement "namespace as tenant" without it unless you know all your namespaces in advance.

**Comment by alanconway** [comment]
@periklis @jcantrill  can I get an LGTM here? This brings the enhancement proposal up to date with the code that merged to the CLO.

**Pull request merged**